packages/dd-trace/src/pkg.js

```diff
@@ -1,17 +1,44 @@
 'use strict'
 
+const fs = require('fs')
 const path = require('path')
-const readPkgUp = require('read-pkg-up')
 
 function findRoot () {
-  return require.main && require.main.filename ? path.dirname(require.main.filename) : process.cwd()
+  return require.main && require.main.filename
+    ? path.dirname(require.main.filename)
+    : process.cwd()
 }
 
 function findPkg () {
   const cwd = findRoot()
-  const up = readPkgUp.sync({ cwd })
+  const filePath = findUp('package.json', cwd)
 
-  return up && up.pkg ? up.pkg : {}
+  try {
+    return readPkg(path.dirname(filePath))
```

Comment:
Why get the dirname if you're just going to convert it right back to the file path again in the function? Why not just have `readPkg` receive the complete file path?

Reply:
Good catch, fixed.